Sources/OpenFeature/Provider/OpenFeatureProvider.swift

```diff
@@ -17,11 +17,11 @@ public protocol OpenFeatureProvider: Service {
     var metadata: OpenFeatureProviderMetadata { get }
     var hooks: [any OpenFeatureHook] { get }
 
-    func resolution(
+    func resolution<Value: OpenFeatureValue>(
```

Comment:
Not sure if a single generic function will work well in the long term or if we should have functions for each value (Bool, String, ...)
But for now this seemed less boilerplate-heavy

Reply:
See my comment below 😊